From e8b8d2f0213eca6129ff3e112626b3096692a688 Mon Sep 17 00:00:00 2001
From: Zhiyi Zhang <zzhang@codeweavers.com>
Date: Thu, 22 Feb 2024 15:45:14 +0800
Subject: [PATCH 1294/1306] fixup! winex11.drv: Listen to RawMotion and
 RawButton* events in the desktop thread.

Fix 7be9a6ec breaks virtual desktop.

CW-Bug-Id: #18383
---
 dlls/winex11.drv/mouse.c       | 5 -----
 dlls/winex11.drv/window.c      | 3 +++
 dlls/winex11.drv/x11drv_main.c | 3 ---
 3 files changed, 3 insertions(+), 8 deletions(-)

diff --git a/dlls/winex11.drv/mouse.c b/dlls/winex11.drv/mouse.c
index 1c853e9393f..befc691263f 100644
--- a/dlls/winex11.drv/mouse.c
+++ b/dlls/winex11.drv/mouse.c
@@ -368,11 +368,6 @@ void X11DRV_XInput2_Enable( Display *display, Window window, long event_mask )
     mask.deviceid = XIAllMasterDevices;
     memset( mask_bits, 0, sizeof(mask_bits) );
 
-    if (NtUserGetWindowThread( NtUserGetDesktopWindow(), NULL ) == GetCurrentThreadId())
-        data->xi2_rawinput_only = TRUE;
-    else
-        data->xi2_rawinput_only = FALSE;
-
     /* FIXME: steam overlay doesn't like if we use XI2 for non-raw events */
 
     if (event_mask & PointerMotionMask)
diff --git a/dlls/winex11.drv/window.c b/dlls/winex11.drv/window.c
index f355bb20c8c..db51152e1dc 100644
--- a/dlls/winex11.drv/window.c
+++ b/dlls/winex11.drv/window.c
@@ -2488,6 +2488,9 @@ BOOL X11DRV_CreateWindow( HWND hwnd )
         struct x11drv_thread_data *data = x11drv_init_thread_data();
         XSetWindowAttributes attr;
 
+        data->xi2_rawinput_only = TRUE;
+        X11DRV_XInput2_Enable( data->display, None, PointerMotionMask | ButtonPressMask | ButtonReleaseMask );
+
         /* create the cursor clipping window */
         attr.override_redirect = TRUE;
         attr.event_mask = StructureNotifyMask | FocusChangeMask;
diff --git a/dlls/winex11.drv/x11drv_main.c b/dlls/winex11.drv/x11drv_main.c
index 2d3104c2641..19c41ea4945 100644
--- a/dlls/winex11.drv/x11drv_main.c
+++ b/dlls/winex11.drv/x11drv_main.c
@@ -932,9 +932,6 @@ struct x11drv_thread_data *x11drv_init_thread_data(void)
     if (use_xim) xim_thread_attach( data );
 
     X11DRV_XInput2_Init();
-    if (NtUserGetWindowThread( NtUserGetDesktopWindow(), NULL ) == GetCurrentThreadId())
-        X11DRV_XInput2_Enable( data->display, None, PointerMotionMask|ButtonPressMask|ButtonReleaseMask );
-
     return data;
 }
 
-- 
2.43.2

